contracts/src/CrossChainApplications/examples/NativeTokenBridge/NativeTokenDestination.sol

```diff
@@ -89,6 +101,10 @@ contract NativeTokenDestination is TeleporterOwnerUpgradeable, INativeTokenDesti
 
         initialReserveImbalance = initialReserveImbalance_;
         currentReserveImbalance = initialReserveImbalance_;
+
+        require(tokenMultiplier_ != 0, "NativeTokenDestination: zero tokenMultiplier");
+        tokenMultiplier = tokenMultiplier_;
```

Comment:
I wonder if it would be clearer for developers if the multiplier was provided as a power of 10 (or even just requiring it to be a power of 10). If we're documenting it's purpose as a means of normalizing the "decimals" count, it wouldn't make sense for it to have any value others than powers of 10 IMO.

Reply:
agreed that it should be in power of 10

Reply:
So I thought about this a bit. I think restricting it to a multiple of 10 seems needlessly restrictive, and would be over-anticipating use cases. I think having the integer multiplier allows the users the most amount of freedom without being confusing. I've updated the docs slightly to allude to other use cases.

Reply:
I'm not sure what you mean by that

Reply:
It isn't specific to non-power-of-10 multipliers, but I'm imagining a case where a multiplier is used to make a token more divisible on the destination because a single unit of it on the source is worth a significant enough amount that the destination subnet wants to allow for smaller subdivisions of it. In that use case, I think our use of integer division is undesirable because if the multiplier is 4 and I move back 7 to the source chain, I only end up with 4.

In the use case of normalizing decimal counts, rounding down to the nearest unit of the source chain seems likely to be negligible (i.e. less than 0.000001 in the case of USDT/USDC), but in the use case of allowing further subdivision of an assets that's not divisible enough on the source, it seems less likely to be negligible.

That being said, I think requiring the amounts sent back to the source chain are divisible by the multiplier could largely avoid this. https://github.com/ava-labs/teleporter/pull/291#discussion_r1504928310